### ISSUE TITLE
not all routes take you to the correct room

### ISSUE THREAD

**Issue by Willow-6855** [HSE-Apps-Team/HseMaps#2]
Example route of B106 to H110 it takes you to H112 instead of H110.

**Comment by RexfordTolle** [comment]
> Example route of B106 to H110 it takes you to H112 instead of H110.

Yeah H110 and H112 are prob assigned to the same node. Will add more nodes later for better precision.

**Comment by RexfordTolle** [comment]
![OIP (1)](https://github.com/user-attachments/assets/c5c3324b-98de-41eb-bf0e-0e45e94fce6b)

**Comment by Willow-6855** [comment]
> > Example route of B106 to H110 it takes you to H112 instead of H110.
> 
> Yeah H110 and H112 are prob assigned to the same node. Will add more nodes later for better precision.

I can add more nodes if need is it just as simple as assigning each room number to single line

**Comment by RexfordTolle** [comment]
> > > Example route of B106 to H110 it takes you to H112 instead of H110.
> 
> > 
> 
> > Yeah H110 and H112 are prob assigned to the same node. Will add more nodes later for better precision.
> 
> 
> 
> I can add more nodes if need is it just as simple as assigning each room number to single line

Not quite

The map consists of several nodes placed at strategic locations (at turns, new hallways, etc). The program simply finds the shortest path from node A to node B through adjacent nodes (nodes that one could travel between without phasing through a wall). Classrooms are manually assigned to the closest node. Right now, there are relatively few nodes so the path finding may be a bit inaccurate.

Unfortunately, adding new nodes manually isn't very easy as changes would have to be made to the distance matrix, vertices, image matrix, and the room-node assignment file.

I am planning on making an algorithm that will take the existing node graph and place an intermediate node between each adjacent node pair to increase precision. Afterwards, it's as simple as assigning the room to the closest node.

Btw, I mistook you for John, apologies for the previous image.
